### PR TITLE
registration spec

### PR DESCRIPTION
**Pull request by aweng98**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: automation/project/Dependencies.scala
Comment (aweng98) [inline automation/project/Dependencies.scala:7]: removing unused akka lib

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala:58]: open windows size must be exact 2. if more, sign of something went wrong.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala]: wait until one condition is true

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala:122]: wait for SignIn popup to go away.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/SignInSpec.scala
Comment (aweng98) [inline automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/user/SignInSpec.scala:26]: I don't think it gains anything in test coverage for login again as user1

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala
Comment (breilly2) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala:58]: Curious why there would be more than 2. Is this something you've seen?

Comment (aweng98) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala:58]: No. a precautionary step.

File: automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala
Comment (breilly2) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala]: This can be achieved with:
```
await condition (id("identifierLink").findElement.exists(_.isDisplayed) || id("identifierId").findElement.exists(_.isEnabled))(10)
```
Using `await condition` has the advantage that it protects against `StaleElementReferenceException` which has a tendency to happen in our app because of re-renders from react. Of course, this case is the Google pop-up so that probably doesn't apply, but `await condition` is also more concise than manually using `WebDriverWait` and `ExpectedConditions` as is more consistent with the rest of the test code.

I dug into the implementations of `visibilityOfElementLocated` and `elementToBeClickable` and they use `isDisplayed` and `isEnabled` respectively so I am confident that the behavior of my suggested code is the same.

Comment (breilly2) [reply automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala]: Also, it looks like this is the real meaningful change in this PR that should help with test reliability, correct? Without seeing it live, I could believe that being more specific about what to look for (visible and clickable instead of just existing) would help so that looks good to me. 👍

**Review by breilly2 — APPROVED** [review]
Just a suggested rephrasing of the await expression and then good to go. 👍 Let me know if you object to the suggested change.

**Comment by aweng98** [comment]
Changed to use `await`. I'm waiting for Jenkin tests to pass before merge. Thank you.

**Pull request merged**
